MovingBlocks/DestinationSol
# Refactor of planet difficulty in JSON

# Description
This PR replaces the `easyOnly` and `hardOnly` flags in `planetsConfig.json` with `difficultyLevels`. What this flag does is enable a planet to be created and placed in all modes. The previous system only allowed a planet to be in easy, medium, or hard mode alone. This is necessary because the game needs to be able to create at least one planet for each game mode. If the planets in the `core` module are disabled -- if you were to remove them from the assets directory for testing or game play reasons -- and you wanted to add a module that only had one planet, (again, for testing or game play) the game would crash because each planet can only be enabled for one mode. The `difficultyLevels` attribute is a list of all difficulties a planet should be spawned in, `"easy"`, `"medium"`, or `"hard"`. Thus allowing a planet to be in all modes, or any subset thereof, so the game doesn't crash.

# Testing
1. Checkout this branch
2. Get @BenjaminAmos' module, blackHole, `./groovyw module get blackHoles -remote BenjaminAmos`.
3. Remove the `grounds` directory from the `core` module, as well as the `planetsConfig.json` and `planetNamesConfig.json` (this step is to force the game to spawn only one type of planet -- black holes)
4. Launch the game, and see the crash.
5. Change the line that reads `"hardOnly" = true` in the `blackHole` module's `planetsConfig.json` to read:
```
"difficultyLevels": [
    "easy",
    "medium",
    "hard"
]
```
6. Relaunch the game and see that all planets are now black holes
